File: Replicon.Cryptography.SCrypt.MMA/SaltParseException.cpp
#include "SaltParseException.h"

namespace Replicon
{
    namespace Cryptography
    {
        namespace SCrypt
        {
            namespace MMA
            {
                SaltParseException::SaltParseException(String^ message)
                    : Exception(message)
                {
                }
            }
        }
    }
}

File: Replicon.Cryptography.SCrypt/SaltParseException.cs
using System;

namespace Replicon.Cryptography.SCrypt
{
    /// <summary>
    /// Exception thrown when a SCrypt salt string is unparsable.
    /// </summary>
    [Serializable]
    public class SaltParseException : Exception
    {
        internal SaltParseException(string message)
            : base(message)
        {
        }
    }
}
